matrixio-core.c

```diff
@@ -103,36 +104,68 @@ int matrixio_hw_reg_write(void *context, unsigned int reg, unsigned int val)
 
 	send_buf[1] = val;
```

Comment:
Isn't there a memory overflow in this line?

send_buf[1] = val;

Also, why isn't there a mutex in this function?